databricks_cli/runs/cli.py

```diff
@@ -39,21 +41,51 @@
               help='File containing JSON request to POST to /api/2.*/jobs/runs/submit.')
 @click.option('--json', default=None, type=JsonClickType(),
               help=JsonClickType.help('/api/2.*/jobs/runs/submit'))
+@click.option('--wait', is_flag=True,
+              help='If specified, the CLI will wait for the submitted run to complete ' +
+                   '(only available in API 2.1).')
 @api_version_option
 @debug_option
 @profile_option
 @eat_exceptions
 @provide_api_client
-def submit_cli(api_client, json_file, json, version):
+def submit_cli(api_client, json_file, json, wait, version):
     """
     Submits a one-time run.
 
     The specification for the request json can be found
     https://docs.databricks.com/api/latest/jobs.html#runs-submit
     """
     check_version(api_client, version)
-    json_cli_base(json_file, json, lambda json: RunsApi(
-        api_client).submit_run(json, version=version))
+    api_version = version or api_client.jobs_api_version
+    if api_version != '2.1' and wait:
+        click.echo(click.style('ERROR', fg='red') + ': the option --wait is only available ' +
+                   'in API 2.1', err=True)
+        return
+
+    if json_file:
+        with open(json_file, 'r') as f:
+            json = f.read()
+    submit_res = RunsApi(api_client).submit_run(json_loads(json), version=version)
+    click.echo(pretty_format(submit_res))
+    if wait:
+        run_id = submit_res['run_id']
+        completed_states = set(['TERMINATED', 'SKIPPED', 'INTERNAL_ERROR'])
+        # Wait for run to complete
+        while True:
+            run = RunsApi(api_client).get_run(run_id, version=version)
+            run_state = run['state']
+            if run_state['life_cycle_state'] in completed_states:
+                if run_state['result_state'] == 'SUCCESS':
+                    click.echo('Job completed successfully.')
+                else:
+                    click.echo(click.style('ERROR', fg='red') + ': job failed with state ' +
+                                run_state['result_state'] + ' and state message ' +
+                                run_state['state_message'], err=True)
+                return
+            click.echo('Job still running with lifecycle state ' + run_state['life_cycle_state'] +
+                        '. URL: ' + run['run_page_url'])
```

Comment:
This seems like debug output. WDYT?

Reply:
This is not. We are following the behavior of the run-notebook open source GitHub action to continuous print the status and link of the job run.

Reply:
@jerrylian-db in the CLI I agree with @pietern that this should be debug output - in the run-notebook Action, we continuously print the URL due to a limitation in GitHub Action output, which is that only output generated after the UI is loaded is shown (so if we don't continuously print output users won't see it if they load UI after the page loads).

Reply:
Either way, please redirect to stderr. If you want to use the `run_id` from the JSON body printed above, you won't be able to if we echo other stuff to stdout.

Reply:
I've redirected this echo to stderr